apps/mainsite/badge_pdf.py

```diff
@@ -0,0 +1,473 @@
+
+font_path_rubik_regular = os.path.join(os.path.dirname(__file__), 'Rubik-Regular.ttf')
+font_path_rubik_medium = os.path.join(os.path.dirname(__file__), 'Rubik-Medium.ttf')
+font_path_rubik_bold = os.path.join(os.path.dirname(__file__), 'Rubik-Bold.ttf')
+
+pdfmetrics.registerFont(TTFont('Rubik-Regular', font_path_rubik_regular))
+pdfmetrics.registerFont(TTFont('Rubik-Medium', font_path_rubik_medium))
+pdfmetrics.registerFont(TTFont('Rubik-Bold', font_path_rubik_bold))
+
+
+class BadgePDFCreator:
+    def __init__(self):
+        pass
+
+    def add_badge_image(self, first_page_content, badgeImage): 
+        image_width = 180
+        image_height = 180
+        first_page_content.append(Image(badgeImage, width=image_width, height=image_height))
+
+    def add_recipient_name(self, first_page_content, name, issuedOn):
+        first_page_content.append(Spacer(1, 58))
+        recipient_style = ParagraphStyle(name='Recipient', fontSize=16, textColor='#492E98',fontName='Rubik-bold',  alignment=TA_CENTER)
+        
+        recipient_name = f"<strong>{name}</strong>"
+        first_page_content.append(Paragraph(recipient_name, recipient_style))
+        first_page_content.append(Spacer(1, 25))
+
+        text_style = ParagraphStyle(name='Text_Style', fontSize=18, alignment=TA_CENTER)
+        
+        text = "hat am " + issuedOn.strftime("%d.%m.%Y")
+        first_page_content.append(Paragraph(text, text_style))
+        first_page_content.append(Spacer(1, 10))
+
+        text = "den folgenden Badge erworben:"
+        first_page_content.append(Paragraph(text, text_style))
+        first_page_content.append(Spacer(1, 25))
+
+
+    def add_title(self, first_page_content, badge_class_name):
+
+        title_style = ParagraphStyle(name='Title', fontSize=20, textColor='#492E98', fontName="Rubik-Bold" , leading=30, alignment=TA_CENTER)
+        first_page_content.append(Paragraph(f"<strong>{badge_class_name}</strong>", title_style))
+        # if(len(badge_class_name) > 30):
+        first_page_content.append(Spacer(1, 15))
+        # else:
+        #     first_page_content.append(Spacer(1, 35))
+
+    def truncate_text(text, max_words=70):
+        words = text.split()
+        if len(words) > max_words:
+            return ' '.join(words[:max_words]) + '...'
+        else:
+            return text
+
+    def add_description(self, first_page_content, description):
+        description_style = ParagraphStyle(name='Description', fontSize=12,fontName='Rubik-Regular',  leading=16.5, alignment=TA_CENTER, leftIndent=20, rightIndent=20)
+        first_page_content.append(Paragraph(description, description_style))
+        first_page_content.append(Spacer(1, 10))
+
+
+    def add_narrative(self, first_page_content, narrative):
+        if narrative is not None:
+            narrative_style = ParagraphStyle(name='Narrative',fontName="Rubik-Italic",fontSize=12,textColor='#6B6B6B', leading=16.5, alignment=TA_CENTER, leftIndent=20, rightIndent=20)
```

Comment:
```Rubik-Italic``` is not a registered font yet, can you add it or use a different font?

Reply:
just added it